src/OneExplorer.ts

```diff
@@ -0,0 +1,196 @@
+      public readonly label: string,
+      public readonly collapsibleState: vscode.TreeItemCollapsibleState,
+      public readonly node: Node,
+  ) {
+    super(label, collapsibleState);
+
+    this.tooltip = `${this.node.uri.fsPath}`;
+
+    if (node.type === NodeType.config) {
+      this.iconPath = new vscode.ThemeIcon('gear');
+    } else if (node.type === NodeType.directory) {
+      this.iconPath = vscode.ThemeIcon.Folder;
+    } else if (node.type === NodeType.model) {
+      this.iconPath = vscode.ThemeIcon.File;
+    }
+  }
+}
+
+export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
+  private _onDidChangeTreeData: vscode.EventEmitter<OneNode|undefined|void> =
+      new vscode.EventEmitter<OneNode|undefined|void>();
+  readonly onDidChangeTreeData: vscode.Event<OneNode|undefined|void> =
+      this._onDidChangeTreeData.event;
+
+  private oneTree: Node|undefined;
+
+  constructor(private workspaceRoot: vscode.Uri|undefined) {
+    if (workspaceRoot !== undefined) {
+      this.oneTree = this.getTree(workspaceRoot);
+    }
+  }
+
+  // TODO(dayo): enable refresh command
+  refresh(): void {
+    this._onDidChangeTreeData.fire();
+  }
+
+  getTreeItem(element: OneNode): vscode.TreeItem {
+    element
+        .command = {command: 'oneExplorer.openFile', title: 'Open File', arguments: [element.node]};
+    return element;
+  }
+
+  getChildren(element?: OneNode): OneNode[]|Thenable<OneNode[]> {
+    if (this.oneTree === undefined) {
+      vscode.window.showInformationMessage('No ONE model or config in empty workspace');
+      return Promise.resolve([]);
+    }
+
+    if (element) {
+      return Promise.resolve(this.getNode(element.node));
+    } else {
+      return Promise.resolve(this.getNode(this.oneTree));
+    }
+  }
+
+  private getNode(node: Node): OneNode[] {
+    const toOneNode = (node: Node): OneNode => {
+      if (node.childNodes.length > 0) {
+        return new OneNode(node.name, vscode.TreeItemCollapsibleState.Collapsed, node);
+      } else {
+        return new OneNode(node.name, vscode.TreeItemCollapsibleState.None, node);
+      }
+    };
+
+    return node.childNodes.map(node => toOneNode(node));
+  }
+
+  private getTree(rootPath: vscode.Uri): Node {
+    const node = {
+      type: NodeType.directory,
+      name: path.parse(rootPath.fsPath).base,
+      dir: true,
+      childNodes: [],
+      uri: rootPath
+    };
+    this.searchNode(node, path.dirname(rootPath.fsPath));
+    return node;
+  }
+
+  private searchNode(node: Node, dirPath: string) {
+    const dirpath = path.join(dirPath, node.name);
+    const files = fs.readdirSync(dirpath);
+
+    for (const fn of files) {
+      const fpath = path.join(dirpath, fn);
+      const fstat = fs.statSync(fpath);
+
+      if (fstat.isDirectory()) {
+        const dirNode = {
+          type: NodeType.directory,
+          name: fn,
+          dir: true,
+          childNodes: [],
+          uri: vscode.Uri.file(fpath)
+        };
```

Comment:
```suggestion
        const dirNode: Node =
            {type: NodeType.directory, name: fn, childNodes: [], uri: vscode.Uri.file(fpath)};
```

I have tested `check format` and it changes to this format.

Reply:
@[jyoungyun](https://github.com/jyoungyun) I have run `infra/format` script locally to resolve our format checker CI. And it has passed the format test.
I think the version of our format tools are different..?

Reply:
What I want to say was that the original code didn't explicitly specify the type of `dirNode`, and in my suggestion, I explicitly named `dirNode` as `Node` type.

The `Node` interface does not have a `dir` property, but there is no error because there is no type of `dirNode`.

Reply:
I caught the point and I have applied it. I just wanted to find out our style linter differences :-D